From b11a801fd67b8940681bd747fde86d90be148b60 Mon Sep 17 00:00:00 2001
From: Shawn Lin <shawn.lin@rock-chips.com>
Date: Mon, 7 Dec 2020 14:09:29 +0800
Subject: [PATCH] mmc: sdhci-of-dwcmshc: Reset the previous DLL settings

Reset the previous DLL settings left out from loader in case
of running initial booting with mismatch DLL settings.

Change-Id: Icba9ddcafc3969f7869a6ba31a017db6be0c885e
Signed-off-by: Shawn Lin <shawn.lin@rock-chips.com>
---
 drivers/mmc/host/sdhci-of-dwcmshc.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/mmc/host/sdhci-of-dwcmshc.c b/drivers/mmc/host/sdhci-of-dwcmshc.c
index b1c5ebba1bac..42bde9787e4c 100644
--- a/drivers/mmc/host/sdhci-of-dwcmshc.c
+++ b/drivers/mmc/host/sdhci-of-dwcmshc.c
@@ -245,6 +245,9 @@ static int rockchip_pltf_init(struct sdhci_host *host, struct dwcmshc_priv *priv
 
 	/* Disable cmd conflict check */
 	sdhci_writel(host, 0x0, DWCMSHC_HOST_CTRL3);
+	/* Reset previous settings */
+	sdhci_writel(host, 0, DWCMSHC_EMMC_DLL_TXCLK);
+	sdhci_writel(host, 0, DWCMSHC_EMMC_DLL_STRBIN);
 	return 0;
 }
 
-- 
2.35.3

